src/value/de.rs

```diff
@@ -304,10 +411,34 @@ impl<'de> serde::Deserializer<'de> for Value {
         visitor.visit_newtype_struct(self)
     }
 
+    #[cfg(not(feature = "arbitrary_precision"))]
+    fn deserialize_struct<V>(self, _name: &'static str, _fields: &'static [&'static str],
+                             visitor: V) -> Result<V::Value, Error>
+        where V: Visitor<'de>,
+    {
+        self.deserialize_any(visitor)
+    }
+
+    #[cfg(feature = "arbitrary_precision")]
+    fn deserialize_struct<V>(self, name: &'static str, fields: &'static [&'static str],
+                             visitor: V) -> Result<V::Value, Error>
+        where V: Visitor<'de>,
+    {
+        if name == SERDE_STRUCT_NAME && fields == &[SERDE_STRUCT_FIELD_NAME] {
```

Comment:
I don't think we need this case here because the Deserialize impls for Number and Value both call deserialize_any, not deserialize_struct (which is fine). AIUI nothing would ever hit this case.

Reply:
This has not been addressed yet.

Reply:
Done.